src/utils/power_flow/make_pf.jl

```diff
@@ -0,0 +1,146 @@
+"""
+    make_pf(sys)
+
+Create the objects needed to solve an powerflow case using NLsolve.jl solvers. Returns 
+an anonymous function with the powerflow equations, initial conditions and a dict to link the
+solutions to the original system. Only supports systems with a single generator per bus and 
+currently doesn't support distributes slack buses and doesn't enforce reactive power limits. 
+
+## Example
+```julia
+pf!, x0, res_ref = make_pf(sys)  
+res = NLsolve.nlsolve(pf!, x0)
+```
+
+# Arguments
+    * `sys`::System : a PowerSystems.jl system
+
+"""
+
+macro fn(expr::Expr)
+    @assert expr.head in (:function, :->)
+    name = gensym()
+    args = expr.args[1]
+    args = typeof(args) == Symbol ? [args] : args.args
+    body = expr.args[2]
+    @eval $name($(args...)) = $body
+    name
+  end
+
+function make_pf(system)
+    buses = sort(collect(get_components(Bus, system)), by = x -> x.number)
+    bus_count = length(buses)
+    #assumes the ordering in YBus is the same as in the buses.
+    Yb = Ybus(system)
+
+    #internally allocate matrices
+    internal = quote
+        V = Vector{ComplexF64}(undef, $bus_count)
+        Vc = Vector{ComplexF64}(undef, $bus_count)
+        P_bal = Vector{Float64}(undef, $bus_count)
+        Q_bal = Vector{Float64}(undef, $bus_count)		
+    end
+
+    var_count = 1
+    res_dict = Dict{String, Vector{Tuple{Symbol, Int}}}() 
+    x0 = Array{Float64}(undef, bus_count*2)	
+
+    for (ix, b) in enumerate(buses)
+        #Gets relevant data about the generators
+        generator = nothing
+        for gen in get_components(Generator, system)
+            if gen.bus == b
+                !isnothing(generator) && throw(DataFormatError("There is more than one generator connected to Bus $b.name"))
+                generator = gen
+            end
+        end        
+        isnothing(generator) ? total_gen = (0.0, 0.0) : total_gen = (generator.activepower, generator.reactivepower)
+        # get load data
+        total_load = [(sum(l.activepower), sum(l.reactivepower)) for l in get_components(ElectricLoad, system) if l.bus == b]
```

Comment:
Are you intending to sum all the activepowers and reactivepowers?  I don't think this does that.  Surprisingly (to me), in Julia sum(3.4) = 3.4

Reply:
yes, that's the intention get the total power at the node.

Reply:
Also, using .activepower means that all ElectricLoad subtypes have to implement this field.  That's why we agreed to use get_activepower() instead.

Reply:
In the new structs it has that field implemented

Reply:
Yeah, but derived structs created by others might not.  That's why we wanted to define the accessor functions as interface methods.

Reply:
FixedAdmittance doesn't have those fields, so it breaks.

Reply:
Ok, I will update that

Reply:
> FixedAdmittance doesn't have those fields, so it breaks. But that's a different problem; it won't get the methods, either.

I didn't consider this case, needs some retooling.